File: regexlib/2021-5-15/javascript_test_file/regexlib_8032.js
/* 8032
 * ^[A-Z]+[A-Z0-9,\x5F]*$
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"A"*10000+"!1 _SLQ_2"
 */
var REGEX = /^[A-Z]+[A-Z0-9,\x5F]*$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'A'.repeat(i*10000) + '!1 _SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_8032.py
# 8032
# ^[A-Z]+[A-Z0-9,\x5F]*$
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"A"*10000+"!1 _SLQ_2"

import re
from time import perf_counter

regex = """^[A-Z]+[A-Z0-9,\x5F]*$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "A" * i * 10000 + "!1 _SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")